### ISSUE TITLE
Item Counts (webui.strengths): Live Counts Break the Community List

### ISSUE THREAD

**Issue by AmyBW** [DSpace/DSpace#9437]
**Describe the bug**
Live item counts prevent the community list from loading in 7.6.1

**To Reproduce**
1. Enable live item counts in the config file (set webui.strengths.show = true and webui.strengths.cache=false) and restart Tomcat.
2. Reload the home page. 
3. Communities & collections will not load, although recent submissions do load as expected. 
- GUI errors include: "Error fetching collection" and "error.top-level-communities". 
- Console and dspace.log state: 
```
java.lang.NullPointerException: Cannot invoke "org.dspace.core.DBConnection.getSession()" because the return value of "org.dspace.core.Context.getDBConnection()" is null
        at org.dspace.core.AbstractHibernateDAO.getHibernateSession(AbstractHibernateDAO.java:59) ~[dspace-api-7.6.1.jar:7.6.1]
        at org.dspace.core.AbstractHibernateDAO.createQuery(AbstractHibernateDAO.java:137) ~[dspace-api-7.6.1.jar:7.6.1]
        at org.dspace.content.dao.impl.ItemDAOImpl.countItems(ItemDAOImpl.java:395) ~[dspace-api-7.6.1.jar:7.6.1]
        at org.dspace.content.ItemServiceImpl.countItems(ItemServiceImpl.java:1589) ~[dspace-api-7.6.1.jar:7.6.1]
        at org.dspace.browse.ItemCounter.getCount(ItemCounter.java:131) ~[dspace-api-7.6.1.jar:7.6.1]
        at org.dspace.content.CommunityServiceImpl.countArchivedItems(CommunityServiceImpl.java:723) ~[dspace-api-7.6.1.jar:7.6.1]
        at org.dspace.content.Community.countArchivedItems(Community.java:275) ~[dspace-api-7.6.1.jar:7.6.1] ...
```
(etc)

**Expected behavior**
Normal home page should load showing top-level communities and accurate item-count badges. 

**Related work**
Cached item counts are inaccurate but the system is still useable: https://github.com/DSpace/DSpace/issues/9434

**Comment by likebirds** [comment]
The other issue is that the count does not 'automatically' count, even after item addition, and for publicly accessible items. Read somewhere that it requires one to restart Tomcat, I have not tried it.

**Comment by AmyBW** [comment]
> The other issue is that the count does not 'automatically' count, even after item addition, and for publicly accessible items. Read somewhere that it requires one to restart Tomcat, I have not tried it.

@likebirds that is correct, and it's filed as a separate issue/bug report: https://github.com/DSpace/DSpace/issues/9434

**Comment by cervacsaba** [comment]
Hi
As webui.strengths.cache=true is not working the way I expected, - I want to have counted and shown the count of all items in every collections, not only the ones with public access (this is the way now that dspace 7 is working now) - I needed to set webui.strengths.cache=false. I got the the same error as mentioned here, but after a while, I found a temporary solution, waiting for the experts to solve both problems a better way. So, I made some changes in 
dspace-api/src/main/java/org/dspace/core/AbstractHibernateDAO.java
checking the exsistense of active db connection and repairing if it is null
/*******************/
import org.dspace.utils.DSpace;
import org.dspace.services.factory.DSpaceServicesFactory;
.
.
private DBConnection dbConnection;


    protected Session getHibernateSession(Context context) throws SQLException {
        if (context.getDBConnection() == null) dbConnection = new DSpace().getServiceManager().getServiceByName(null, DBConnection.class);
        else dbConnection = context.getDBConnection();
        return ((Session) dbConnection.getSession());
    }
 /***************/

**Comment by alanorth** [comment]
@AmyBW thank you for the bug report! I can reproduce it locally on DSpace 7.6.1 as well. When `webui.strengths.cache = false` in `dspace.cfg` the frontend is completely broken.

Community list never loads:

![Screenshot 2024-04-15 at 15-54-31 CGSpace Community List](https://github.com/DSpace/DSpace/assets/191754/e15b4c2b-3bbd-475e-93f7-bd3d75aeac46)

Community page throws the error "Error fetching community":

![Screenshot 2024-04-15 at 15-54-45 DSpace](https://github.com/DSpace/DSpace/assets/191754/0b7ca633-8721-41b2-bc63-f6b9c30c2c71)

And collection pages give an obscure "No item found for the identifier" error:

![Screenshot 2024-04-15 at 15-56-31 DSpace](https://github.com/DSpace/DSpace/assets/191754/663a86eb-2f16-4ac1-b429-a3e7ae023ed8)

**Comment by tdonohue** [comment]
Fix for this issue now available in #9583.  This PR is under review for 8.0 and 7.6.2.  If anyone wants to help test it, I'd appreciate it.
